rclcpp/topics/minimal_subscriber/CMakeLists.txt

```diff
@@ -32,6 +33,26 @@ ament_target_dependencies(subscriber_member_function_with_unique_network_flow_en
 add_executable(subscriber_not_composable not_composable.cpp)
 ament_target_dependencies(subscriber_not_composable rclcpp std_msgs)
 
+add_library(wait_set_subscriber SHARED wait_set_subscriber.cpp)
+ament_target_dependencies(wait_set_subscriber rclcpp rclcpp_components std_msgs)
+rclcpp_components_register_nodes(wait_set_subscriber WaitSetSubscriber)
```

Comment:
Do you get an executable for each of these? If not, then I think you want `rclcpp_compnents_register_node()`? That way you can specify an executable name with `EXECUTABLE <name>`. For example:

https://github.com/ros2/demos/blob/3af19c93c7abd9a83caca47de0780e941faeb22b/demo_nodes_cpp/CMakeLists.txt#L82-L84

The plural one is for libraries that contain more than one node, see this presentation (search for `rclcpp_components_register_node`): https://roscon.ros.org/2019/talks/roscon2019_composablenodes.pdf

Reply:
I moved all the `cpp` files to one library and now I'm using `rclcpp_compnents_register_node` to register each component individually. I updated the `CMakeLists.txt` from wait_set package too.